fluent/src/bundle.js

```diff
@@ -170,74 +161,80 @@ export default class FluentBundle {
-        if (allowOverrides === false && this._terms.has(id)) {
-          errors.push(`Attempt to override an existing term: "${id}"`);
+        if (allowOverrides === false && this._terms.has(entry.id)) {
+          errors.push(`Attempt to override an existing term: "${entry.id}"`);
           continue;
         }
-        this._terms.set(id, value);
+        this._terms.set(entry.id, entry);
       } else {
-        if (allowOverrides === false && this._messages.has(id)) {
-          errors.push(`Attempt to override an existing message: "${id}"`);
+        if (allowOverrides === false && this._messages.has(entry.id)) {
+          errors.push(`Attempt to override an existing message: "${entry.id}"`);
           continue;
         }
-        this._messages.set(id, value);
+        this._messages.set(entry.id, entry);
       }
     }
 
     return errors;
   }
 
   /**
-   * Format a message to a string or null.
+   * Format a Pattern to a string.
    *
-   * Format a raw `message` from the bundle into a string (or a null if it has
-   * a null value).  `args` will be used to resolve references to variables
-   * passed as arguments to the translation.
+   * Format a raw `Pattern` into a string. `args` will be used to resolve
+   * references to variables passed as arguments to the translation.
    *
-   * In case of errors `format` will try to salvage as much of the translation
-   * as possible and will still return a string.  For performance reasons, the
-   * encountered errors are not returned but instead are appended to the
-   * `errors` array passed as the third argument.
+   * In case of errors `formatPattern` will try to salvage as much of the
+   * translation as possible and will still return a string. For performance
+   * reasons, the encountered errors are not returned but instead are appended
+   * to the `errors` array passed as the third argument.
    *
    *     const errors = [];
    *     bundle.addMessages('hello = Hello, { $name }!');
    *     const hello = bundle.getMessage('hello');
-   *     bundle.format(hello, { name: 'Jane' }, errors);
-   *
-   *     // Returns 'Hello, Jane!' and `errors` is empty.
-   *
-   *     bundle.format(hello, undefined, errors);
-   *
-   *     // Returns 'Hello, name!' and `errors` is now:
-   *
-   *     [<ReferenceError: Unknown variable: name>]
-   *
-   * @param   {Object | string}    message
-   * @param   {Object | undefined} args
-   * @param   {Array}              errors
-   * @returns {?string}
+   *     if (hello.value) {
+   *         bundle.formatPattern(hello.value, { name: 'Jane' }, errors);
+   *         // Returns 'Hello, Jane!' and `errors` is empty.
+   *
+   *         bundle.formatPattern(hello.value, undefined, errors);
+   *         // Returns 'Hello, {$name}!' and `errors` is now:
+   *         // [<ReferenceError: Unknown variable: name>]
+   *     }
+   *
+   * @param   {string|Array} pattern
+   * @param   {?Object} args
+   * @param   {?Array} errors
```

Comment:
I'm a bit concerned about `errors` being optional. I understand that it's nice for tests, but it's really not how the API should be used.

Interestingly, `args` is actually optional in the code.

I'd feel better about this if we created an empty array in `scope` if no errors are given.

Maybe this is good for a follow-up? I.e., declaring them optional or not, and how to deal with errors when they're not passed in? (Sneak peek, I wonder if we should throw with the actual error instead of with `undefined has no method push`.

Reply:
Could leaving `errors` undefined mean that the error gets thrown instead?

Reply:
> I'd feel better about this if we created an empty array in `scope` if no errors are given.

That's how it works right now. It happens in `_createScope(args, errors = []) {..}`.

Reply:
oops. Filed #390 on making errors actually fail verbosely if they're hidden.

Reply:
Ah, I see what you mean. Thanks for filing #390.